libkmod/libkmod-index.c

```diff
@@ -183,14 +183,15 @@ static int read_char(FILE *in)
 	return ch;
 }
 
-static uint32_t read_long(FILE *in)
+static int read_long(FILE *in, uint32_t *l)
```

Comment:
no need to change in this patch though.

Reply:
In fact it is needed, because otherwise I cannot return an error value. I would have to switch from `uint32_t` to something like `int64_t` just for a proper negative return value, messing up further type checks and variables which actually expected a `uint32_t` type.

Changed function name to `read_u32` though.

Reply:
I was only referring to read_**long** returning a 32-bit type. Using int as return and passing a pointer is better indeed than returning int64_t.